### PR TITLE
Abbreviate "parameter" -> "param"

### PR DESCRIPTION
**Pull request by josh11b**
Part of switching to the [abbreviations we've decided to use](https://docs.google.com/document/d/1RRYMm42osyqhI2LyjrjockYCutQ5dOf8Abu50kTrkX0/edit?resourcekey=0-kHyqOESbOHmzZphUbtLrTw#heading=h.pph7i5m5un7q).

I will rename files in a follow-up PR.

### CODE REVIEW

File: toolchain/sem_ir/typed_insts.h [hunk 1]
```diff
@@ -335,8 +335,8 @@ struct NoOp {
   // This instruction doesn't produce a value, so has no type.
 };
 
-struct Parameter {
-  static constexpr auto Kind = InstKind::Parameter.Define("parameter");
+struct Param {
+  static constexpr auto Kind = InstKind::Param.Define("parameter");
```
Comment (jonmeow) [inline toolchain/sem_ir/typed_insts.h]:
```suggestion
  static constexpr auto Kind = InstKind::Param.Define("param");
```

Comment (jonmeow) [reply toolchain/sem_ir/typed_insts.h]: This should probably change. It'll cause a lot of test edits, but I don't think those need to be double-checked.

**Review by jonmeow — APPROVED** [review]
Looks good, thanks!

File: toolchain/sem_ir/typed_insts.h (hunk 1 above)
Comment (josh11b) [reply toolchain/sem_ir/typed_insts.h]: It doesn't cause any test edits.

**Pull request merged**